File: frontend/src/types/case.ts
import type { Scene } from './scene'
import type { Evidence } from './evidence'
import type { Suspect } from './suspect'

export type Difficulty = 'Easy' | 'Medium' | 'Hard'
export type CaseStatus = 'Available' | 'In Progress' | 'Completed' | 'Draft' | 'Published'


export interface Case {
  id: string
  caseNumber: string
  title: string
  description: string
  difficulty: Difficulty
  location: string
  time: string
  objective: string
  initialInfo: string
  status: CaseStatus
  coverImage?: string
  initialSceneId: string
  scenes: Scene[]
  evidences: Evidence[]
  suspects: Suspect[]
  solution: {
    correctSuspectId: string
    keyEvidenceIds: string[]
    explanationKeywords: string[]
  }
}

File: backend/app/models/case.py
from datetime import datetime

class CaseModel:
    """Model Kasus Investigasi"""
    def __init__(self, id, title, description, difficulty='Medium', location='', objective='', status='Available'):
        self.id = id
        self.title = title
        self.description = description
        self.difficulty = difficulty
        self.location = location
        self.objective = objective
        self.status = status
        self.created_at = datetime.utcnow()

    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'difficulty': self.difficulty,
            'location': self.location,
            'objective': self.objective,
            'status': self.status,
            'created_at': self.created_at.isoformat()
        }
